Active STN integration branch

msheiny opened the pull request:
Leaving this open for up-to-date peer review and collaboration.

msheiny committed:
Made the application code and npm directory extensible

This allows pulling git repos that do not store the npm and django
app logic at the root folder level.

msheiny committed:
Remove hardcoded python3 virtualenv preference

conorsch commented on defaults/main.yml:
```diff
@@ -11,6 +12,7 @@ django_stack_gcorn_port: 8000
 django_stack_gcorn_log: /var/log/{{django_stack_app_name}}/app
 django_stack_gcorn_app: freedompress.wsgi
 django_stack_gcorn_app_settings: freedompress.settings.production
```
Seems like these vars would be better structured as `{{ django_stack_app_name }}.wsgi` and `{{ django_stack_app_name }}.settings.production`. Might still require overrides, but makes more sense to infer namespace.

msheiny replied:
Something like this is a good compromise?

``` yaml
 django_stack_gcorn_app: "{{ global_app_name }}.wsgi"
 django_stack_gcorn_app_settings: "{{ global_app_name }}.settings.production"
```

msheiny replied:
oh wait

msheiny replied:
hahaha

conorsch replied:
Sure, as long as the "global" var is also namedspace, e.g. `{{ django_stack_global_app_name }}`.

msheiny replied:
Okay I added what you pointed out in latest commit. :+1: Good looking out buddy

msheiny committed:
Added optional global pip package

Also expanded virtualenv pip install to allow pulling in site packages.

msheiny committed:
Little comment cleanup of defaults

msheiny committed:
Expand npm installation. Allow optional config via vars

msheiny committed:
Added optional shell commands

msheiny committed:
Lets wax these environment variables

msheiny committed:
Added git change triggers to some of the command tasks

msheiny committed:
Standardize django app settings variable string

msheiny committed:
Tweaked default vars to reduce override requirements

Particularly for django_stack_gcorn_app and django_stack_gcorn_app_settings
so they will now embed the defined `django_stack_app_name`. TL;DR there is a
better chance that playbooks will have to override less default variables.

msheiny committed:
Fixed environment export setting for virtualenv

This is a band-aid for a fix for ensuring environment variables
are exported properly when directly activating the virtualenv. The issue
is that the gunicorn job doesnt actually activate the virtualenv but
is calling the gunicorn binary from within the env folder. I had to
utilize the EnvironmentFile parameter in systemd which did not take in
variable lines that had `export` in front of them.
